File: tests/ascii_torus/test_metrics.py
import pytest
from mcik.experiments.ascii_torus.metrics import estimate_ascii_quality

def test_ascii_quality_deterministic():
    w, h = 8, 4
    buf = [" "] * (w * h)
    for x in range(w):
        buf[(h // 2) * w + x] = "#"
    q1 = estimate_ascii_quality(buf, w, h)
    q2 = estimate_ascii_quality(buf, w, h)
    assert q1 == pytest.approx(q2)

File: tests/ascii_torus/test_metrics.cpp
#define CATCH_CONFIG_MAIN
#include <catch2/catch.hpp>
#include <vector>
#include "mcik/experiments/ascii_torus/metrics.hpp"

TEST_CASE("ASCII quality estimator is deterministic on fixed buffer") {
  const int w = 8, h = 4;
  std::vector<char> buf(w * h, ' ');
  for (int x = 0; x < w; ++x) buf[(h / 2) * w + x] = '#';
  double q1 = torus_metrics::estimate_ascii_quality(buf, w, h);
  double q2 = torus_metrics::estimate_ascii_quality(buf, w, h);
  REQUIRE(q1 == Approx(q2));
}
